src/styles/blocks/card.scss

```diff
@@ -0,0 +1,104 @@
+.card {
+  display: flex;
+  flex-direction: column;
+  flex-grow: 0;
+  width: 200px;
+  padding: 16px;
+  margin-right: 48px;
+  margin-bottom: 46px;
+
+  background: $white;
+  border: 1px solid $grayElements;
+  border-radius: 5px;
+
+  &__image {
+    height: 134px;
+    width: 160px;
+    margin: auto;
+    margin-top: 16px;
+    margin-bottom: 40px;
+  }
+
+  &__product-name {
+    padding-bottom: 4px;
+
+    font-style: normal;
+    font-weight: 500;
+    line-height: 18px;
+    font-size: 12px;
+    color: $mainAccent;
+  }
+
+  &__product-code {
+    padding-bottom: 16px;
+
+    font-style: normal;
+    font-weight: 400;
+    line-height: 14px;
+    font-size: 10px;
+    color: #616070;
+  }
+
+  &__price {
+    display: flex;
+    justify-content: space-between;
+
+    padding-bottom: 16px;
+  }
+
+  &__price-text {
+    display: flex;
+    align-items: flex-end;
+    flex: none;
+    order: 0;
+    flex-grow: 0;
+
+    font-style: normal;
+    font-weight: 400;
+    font-size: 12px;
+    line-height: 18px;
+    color: $secondary;
+  }
+
+  &__price-value {
+    display: flex;
+    align-items: flex-end;
+    text-align: right;
+    flex: none;
+    order: 1;
+    flex-grow: 0;
+
+    font-style: normal;
+    font-weight: 700;
+    font-size: 16px;
+    line-height: 18px;
+    color: $mainAccent;
+  }
+
+  &__buy-button {
+    font-style: normal;
+    font-weight: 700;
+    font-size: 14px;
+
+    line-height: 40px;
+    height: 40px;
+    text-align: center;
+    text-decoration: none;
+    text-transform: uppercase;
+    color: $white;
+
+    background-color: $blueAccent;
+    border: 1px solid $blueAccent;
+    border-radius: 5px;
+
+    transition: color 0.3s, background-color 0.3s;
+  }
+
+  &__buy-button:hover {
+    color: $blueAccent;
+
+    background-color: $white;
+    border: 1px solid $blueAccent;
```

Comment:
You shouldn't set new border on hover state, you should change only border color